Module 5 Exercise
Ansel Gomez
Section 2

In [96]:
import pandas as pd
import numpy as np


Read the data movies dataset from this link (Links to an external site.) into a Pandas dataframe. How many movies are there in the dataset? (1 point)

In [97]:
df = pd.read_json("https://raw.githubusercontent.com/vega/vega-datasets/master/data/movies.json") 
df

Title    US Gross  Worldwide Gross  US DVD Sales  \
0                 The Land Girls    146083.0         146083.0           NaN   
1         First Love, Last Rites     10876.0          10876.0           NaN   
2     I Married a Strange Person    203134.0         203134.0           NaN   
3           Let's Talk About Sex    373615.0         373615.0           NaN   
4                           Slam   1009819.0        1087521.0           NaN   
...                          ...         ...              ...           ...   
3196  Zack and Miri Make a Porno  31452765.0       36851125.0    21240321.0   
3197                      Zodiac  33080084.0       83080084.0    20983030.0   
3198                        Zoom  11989328.0       12506188.0     6679409.0   
3199         The Legend of Zorro  45575336.0      141475336.0           NaN   
3200           The Mask of Zorro  93828745.0      233700000.0           NaN   

      Production Budget Release Date MPAA Rating  Running Time min  \
0             8000000.0  Jun 12 1998           R               NaN   
1              300000.0  Aug 07 1998           R               NaN   
2              250000.0  Aug 28 1998        None               NaN   
3              300000.0  Sep 11 1998        None               NaN   
4             1000000.0  Oct 09 1998           R               NaN   
...                 ...          ...         ...               ...   
3196         24000000.0  Oct 31 2008           R             101.0   
3197         85000000.0  Mar 02 2007           R             157.0   
3198         35000000.0  Aug 11 2006          PG               NaN   
3199         80000000.0  Oct 28 2005          PG             129.0   
3200         65000000.0  Jul 17 1998       PG-13             136.0   

             Distributor                        Source        Major Genre  \
0               Gramercy                          None               None   
1                 Strand                          None              Drama   
2              Lionsgate                          None             Comedy   
3              Fine Line                          None             Comedy   
4                Trimark           Original Screenplay              Drama   
...                  ...                           ...                ...   
3196       Weinstein Co.           Original Screenplay             Comedy   
3197  Paramount Pictures     Based on Book/Short Story  Thriller/Suspense   
3198       Sony Pictures  Based on Comic/Graphic Novel          Adventure   
3199       Sony Pictures                        Remake          Adventure   
3200       Sony Pictures                        Remake          Adventure   

             Creative Type         Director  Rotten Tomatoes Rating  \
0                     None             None                     NaN   
1                     None             None                     NaN   
2                     None             None                     NaN   
3                     None             None                    13.0   
4     Contemporary Fiction             None                    62.0   
...                    ...              ...                     ...   
3196  Contemporary Fiction      Kevin Smith                    65.0   
3197         Dramatization    David Fincher                    89.0   
3198            Super Hero     Peter Hewitt                     3.0   
3199    Historical Fiction  Martin Campbell                    26.0   
3200    Historical Fiction  Martin Campbell                    82.0   

      IMDB Rating  IMDB Votes  
0             6.1      1071.0  
1             6.9       207.0  
2             6.8       865.0  
3             NaN         NaN  
4             3.4       165.0  
...           ...         ...  
3196          7.0     55687.0  
3197          NaN         NaN  
3198          3.4      7424.0  
3199          5.7     21161.0  
3200          6.7      4789.0  

[3201 rows x 16 columns]

There are 3201 movies.

Drop all columns except 'Title', 'US Gross', 'Worldwide Gross', ‘US DVD Sales’, 'Production Budget', and ’IMDB Rating'. (1 point)

In [98]:
df = df.drop(columns=['Release Date', 'MPAA Rating', 'Running Time min', 'Distributor', 'Source', 'Major Genre', 'Creative Type', 'Director', 'Rotten Tomatoes Rating', 'IMDB Votes'])
df

Title    US Gross  Worldwide Gross  US DVD Sales  \
0                 The Land Girls    146083.0         146083.0           NaN   
1         First Love, Last Rites     10876.0          10876.0           NaN   
2     I Married a Strange Person    203134.0         203134.0           NaN   
3           Let's Talk About Sex    373615.0         373615.0           NaN   
4                           Slam   1009819.0        1087521.0           NaN   
...                          ...         ...              ...           ...   
3196  Zack and Miri Make a Porno  31452765.0       36851125.0    21240321.0   
3197                      Zodiac  33080084.0       83080084.0    20983030.0   
3198                        Zoom  11989328.0       12506188.0     6679409.0   
3199         The Legend of Zorro  45575336.0      141475336.0           NaN   
3200           The Mask of Zorro  93828745.0      233700000.0           NaN   

      Production Budget  IMDB Rating  
0             8000000.0          6.1  
1              300000.0          6.9  
2              250000.0          6.8  
3              300000.0          NaN  
4             1000000.0          3.4  
...                 ...          ...  
3196         24000000.0          7.0  
3197         85000000.0          NaN  
3198         35000000.0          3.4  
3199         80000000.0          5.7  
3200         65000000.0          6.7  

[3201 rows x 6 columns]

Set 'Title' as the index of your dataframe. (1 point)

In [99]:
df = df.set_index('Title')
df

US Gross  Worldwide Gross  US DVD Sales  \
Title                                                                   
The Land Girls                146083.0         146083.0           NaN   
First Love, Last Rites         10876.0          10876.0           NaN   
I Married a Strange Person    203134.0         203134.0           NaN   
Let's Talk About Sex          373615.0         373615.0           NaN   
Slam                         1009819.0        1087521.0           NaN   
...                                ...              ...           ...   
Zack and Miri Make a Porno  31452765.0       36851125.0    21240321.0   
Zodiac                      33080084.0       83080084.0    20983030.0   
Zoom                        11989328.0       12506188.0     6679409.0   
The Legend of Zorro         45575336.0      141475336.0           NaN   
The Mask of Zorro           93828745.0      233700000.0           NaN   

                            Production Budget  IMDB Rating  
Title                                                       
The Land Girls                      8000000.0          6.1  
First Love, Last Rites               300000.0          6.9  
I Married a Strange Person           250000.0          6.8  
Let's Talk About Sex                 300000.0          NaN  
Slam                                1000000.0          3.4  
...                                       ...          ...  
Zack and Miri Make a Porno         24000000.0          7.0  
Zodiac                             85000000.0          NaN  
Zoom                               35000000.0          3.4  
The Legend of Zorro                80000000.0          5.7  
The Mask of Zorro                  65000000.0          6.7  

[3201 rows x 5 columns]

Assume return on investment (ROI) for a movie is defined as:
𝑅𝑂𝐼 = 𝑊𝑜𝑟𝑙𝑑𝑤𝑖𝑑𝑒 𝑔𝑟𝑜𝑠𝑠 − 𝑃𝑟𝑜𝑑𝑢𝑐𝑡𝑖𝑜𝑛 𝑏𝑢𝑑𝑔𝑒𝑡
What is the mean ROI? Which movie has the maximum ROI? Is it the one with the minimum production budget? (2 points)

In [100]:
roi = df['Worldwide Gross'] - df['Production Budget']
print(f"The mean ROI is {roi.mean()}")
roi

The mean ROI is 54228551.78202318


Title
The Land Girls                 -7853917.0
First Love, Last Rites          -289124.0
I Married a Strange Person       -46866.0
Let's Talk About Sex              73615.0
Slam                              87521.0
                                 ...     
Zack and Miri Make a Porno     12851125.0
Zodiac                         -1919916.0
Zoom                          -22493812.0
The Legend of Zorro            61475336.0
The Mask of Zorro             168700000.0
Length: 3201, dtype: float64

In [101]:
print(f"The movie that has the maximum ROI is {roi.idxmax()} with an ROI of {roi.max()}")

The movie that has the maximum ROI is Avatar with an ROI of 2530891499.0


In [102]:
if roi.idxmax() == df['Production Budget'].idxmin():
    print("The movie with the max ROI is the same as the one with the minimum production budget")
else:
    print("The movie with the max ROI is not the same as the one with the minimum production budget")

The movie with the max ROI is not the same as the one with the minimum production budget


What is the ROI of the movie with the maximum US DVD sales? Is it in the top 25% movies in terms of ROI? Is it in the bottom 25% movies in terms of production budget? (2 points)

In [103]:
print(f"The ROI of {df['US DVD Sales'].idxmax()} is {roi[df['US DVD Sales'].idxmax()]}")

The ROI of The Chronicles of Narnia: The Lion, the Witch and the Wardrobe is 568806957.0


In [104]:
if roi[df['US DVD Sales'].idxmax()] >= roi.describe()['75%']:
    print(f"Yes, it is in the top 25% in terms of ROI")
else:
    print(f"No, it is not in the top 25% in terms of ROI")

Yes, it is in the top 25% in terms of ROI


In [105]:
if df['Production Budget'][df['US DVD Sales'].idxmax()] <= df['Production Budget'].describe()['25%']:
    print(f"Yes, it is in the bottom 25% in terms of Production Budget")
else:
    print(f"No, it is not in the bottom 25% in terms of Production Budget")

No, it is not in the bottom 25% in terms of Production Budget


What is the median difference between worldwide gross and US gross? Which movie has the highest difference? (1 point)

In [106]:
median_gross = df['Worldwide Gross'].median() - df['US Gross'].median()

In [107]:
print(f"The median difference between worldwide gross and US gross is {median_gross}")

The median difference between worldwide gross and US gross is 9149461.0


In [108]:
gross_diff = df['Worldwide Gross'] - df['US Gross']

In [109]:
print(f"The movie with the highest difference is {gross_diff.idxmax()} with a difference of {gross_diff.max()}")

The movie with the highest difference is Avatar with a difference of 2007723849.0


Do a majority of the movies have IMDB ratings higher than the average? Justify your answer. (2 points)

In [110]:
counter = 0

In [111]:
for element in df['IMDB Rating']:
    if element >= df['IMDB Rating'].mean():
        counter += 1

In [112]:
print(counter)

1647


In [121]:
half = (len(df) / 2) + 1

In [122]:
majority = np.ceil(half)

In [127]:
print(f"The amount of movies with a IMDB rating higher than the average is {counter}, which is more than half the movies.")

The amount of movies with a IMDB rating higher than the average is 1647, which is more than half the movies.


In [126]:
print(f"The majority being more than {majority}.")

The majority being more than 1602.0.


Github link

https://github.com/amg515/iti221-excercises